src/BuildScriptGenerator/DefaultCompatiblePlatformDetector.cs

```diff
@@ -34,13 +36,34 @@ public class DefaultCompatiblePlatformDetector : ICompatiblePlatformDetector
 
         /// <inheritdoc/>
         public IDictionary<IProgrammingPlatform, string> GetCompatiblePlatforms(RepositoryContext ctx)
+        {
+            return GetCompatiblePlatforms(ctx, detectionResults: null, runDetection: true);
+        }
+
+        /// <inheritdoc/>
+        public IDictionary<IProgrammingPlatform, string> GetCompatiblePlatforms(
+            RepositoryContext ctx,
+            IEnumerable<PlatformDetectorResult> detectionResults)
+        {
+            return GetCompatiblePlatforms(ctx, detectionResults, runDetection: false);
+        }
```

Comment:
I see here makes sense.. please ignore previous comment. The name may be a little confusing

Reply:
Thanks @qianz2 .Yeah basically I want to avoid running the detection again if it was already run earlier. Ideally I want to refactor this logic to fix the layering, but since that is going to take more time and might conflict with what you are currently doing, this is a temporary solution for now.